lib/std/http/Server.zig

```diff
@@ -336,42 +334,92 @@ pub const Response = struct {
     headers: http.Headers,
     request: Request,
 
+    state: State = .first,
+
+    const State = enum {
+        first,
+        start,
+        waited,
+        responded,
+        finished,
+    };
+
     pub fn deinit(res: *Response) void {
-        res.server.allocator.destroy(res);
-    }
+        res.connection.close();
 
-    /// Reset this response to its initial state. This must be called before handling a second request on the same connection.
-    pub fn reset(res: *Response) void {
-        res.request.headers.deinit();
         res.headers.deinit();
+        res.request.headers.deinit();
 
-        switch (res.request.compression) {
-            .none => {},
-            .deflate => |*deflate| deflate.deinit(),
-            .gzip => |*gzip| gzip.deinit(),
-            .zstd => |*zstd| zstd.deinit(),
+        if (res.request.parser.header_bytes_owned) {
+            res.request.parser.header_bytes.deinit(res.server.allocator);
+        }
+    }
+
+    /// Reset this response to its initial state. This must be called before handling a second request on the same connection.
+    pub fn reset(res: *Response) bool {
```

Comment:
Could this return an `enum { alive, done }` (or similar) instead? Functions which aren't of the form `isX`, `hasX`, `conainsX` have a non obvious result when they return bool. The tag being `.alive`, `.kept_alive`, or whatever doesn't matter as much as it being anything other than `true`/`false` as those don't communicate as much.

see https://github.com/ziglang/zig/issues/12906